lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+module Travis
+  module Build
+    class Script
+      class R < Script
+        DEFAULTS = {
+          # Basic config options
+          cran: 'http://cran.rstudio.com',
+          warnings_are_errors: false,
+          # Dependencies (installed in this order)
+          apt_packages: [],
+          r_binary_packages: [],
+          r_packages: [],
+          bioc_packages: [],
+          r_github_packages: [],
+          # Build/test options
+          r_build_args: '--no-build-vignettes --no-manual',
+          r_check_args: '--no-build-vignettes --no-manual --as-cran',
+          r_check_cran_incoming: false,
+          r_check_revdep: false,
+          # Heavy dependencies
+          latex: false,
+          pandoc: false,
+          pandoc_version: '1.12.4.2',
+          # Bioconductor
+          bioc: 'http://bioconductor.org/biocLite.R',
+          bioc_required: false,
+          bioc_use_devel: true,
+        }
+
+        def initialize(data)
+          # TODO(craigcitro): Is there a way to avoid explicitly
+          # naming arguments here?
+          super
+          @devtools_installed = false
+          @bioc_installed = false
+        end
+
+        def export
+          super
+          sh.export 'TRAVIS_R_VERSION', 'release', echo: false
+        end
+
+        def setup
+          super
+
+          sh.echo 'R for Travis-CI is not officially supported, ' +
+                  'but is community maintained.', ansi: :green
+          sh.echo 'Please file any issues using the following link',
+                  ansi: :green
+          sh.echo '  https://github.com/travis-ci/travis-ci/issues' +
+                  '/new?labels=community:r', ansi: :green
+          sh.echo 'and mention \`@craigcitro\`, \`@eddelbuettel\` and ' +
+                  '\`@hadley\` in the issue', ansi: :green
+
+          # TODO(craigcitro): python-software-properties?
+          sh.echo 'Installing R'
+          case config[:os]
+          when 'linux'
+            # Set up our CRAN mirror.
+            sh.cmd 'sudo add-apt-repository ' +
+                   "\"deb #{config[:cran]}/bin/linux/ubuntu " +
+                   "$(lsb_release -cs)/\""
+            sh.cmd 'sudo apt-key adv --keyserver keyserver.ubuntu.com ' +
+                   '--recv-keys E084DAB9'
```

Comment:
Until the Ubuntu / Canonical Launchpad.net changes. no need.  Constant is fine.

Just this week I reactivated a PPA created in 2009.   Seems pretty "constant" to me.